src/install.c

```diff
@@ -1159,6 +1159,85 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* bu
 	return res;
 }
 
+static gchar* get_uptime(void)
+{
+	gchar *contents = NULL;
+	g_autoptr(GError) ierror = NULL;
+
+	if (!g_file_get_contents("/proc/uptime", &contents, NULL, &ierror)) {
+		g_warning("Failed to get uptime: %s", ierror->message);
+		return NULL;
+	}
+
+	/* file contains newline, modify in-place */
+	return g_strchomp(contents);
+}
+
+/* If the input key starts with RAUC_HTTP_, it returns a valid HTTP header
+ * string with 'RAUC_HTTP_' replaced by 'X-RAUC-'.
+ * If the input string does not start with RAUC_HTTP_, NULL is returned.
+ */
+static gchar *system_info_to_header(const gchar *key, const gchar *value)
+{
+	g_autofree gchar *header_key = NULL;
+
+	g_return_val_if_fail(key, NULL);
+	g_return_val_if_fail(value, NULL);
+
+	if (!g_str_has_prefix(key, "RAUC_HTTP_"))
+		return NULL;
+
+	header_key = g_strdup(key + strlen("RAUC_HTTP_"));
+	for (size_t i = 0; i < strlen(header_key); i++) {
+		if (header_key[i] == '_')
+			header_key[i] = '-';
+	}
+
+	return g_strdup_printf("X-RAUC-%s: %s", header_key, value);
+}
+
+static gchar **assemble_info_headers(RaucInstallArgs *args)
+{
+	GPtrArray *headers = g_ptr_array_new_full(1, g_free);
```

Comment:
`_new_full` with a fixed, low size isn't really useful. Perhaps just use `g_ptr_array_new_with_free_func`, as this function isn't called often.